components/authz-service/server/v2/rules_test.go

```diff
@@ -143,6 +144,171 @@ func TestCreateRule(t *testing.T) {
 	}
 }
 
+func TestGetRule(t *testing.T) {
+	ctx := context.Background()
+	cl, store, _, _ := setupRules(t)
+
+	apiConditions := []*api.Condition{
+		{
+			Type:   api.ProjectRuleConditionTypes_CHEF_ORGS,
+			Values: []string{"opscode"},
+		},
+	}
+	storageConditions := []storage.Condition{
+		{
+			Type:      storage.Node,
+			Attribute: storage.Organization,
+			Operator:  storage.MemberOf,
+			Value:     []string{"opscode"},
+		},
+	}
+
+	cases := []struct {
+		desc string
+		f    func(*testing.T)
+	}{
+		{"if the rule id is empty, returns 'invalid argument'", func(t *testing.T) {
+			resp, err := cl.GetRule(ctx, &api.GetRuleReq{Id: ""})
+			grpctest.AssertCode(t, codes.InvalidArgument, err)
+			assert.Nil(t, resp)
+		}},
+		{"if the rule id is invalid, returns 'invalid argument'", func(t *testing.T) {
+			resp, err := cl.GetRule(ctx, &api.GetRuleReq{Id: "no_underscore_allowed"})
+			grpctest.AssertCode(t, codes.InvalidArgument, err)
+			assert.Nil(t, resp)
+		}},
+		{"if the rule does not exist, returns code 'not found'", func(t *testing.T) {
+			resp, err := cl.GetRule(ctx, &api.GetRuleReq{Id: "foo"})
+			grpctest.AssertCode(t, codes.NotFound, err)
+			assert.Nil(t, resp)
+		}},
+		{"if a rule with the requested id exists, returns the rule", func(t *testing.T) {
```

Comment:
From the name, it is not clear how this test differs from the next; suggest rename to "if there is only one rule and it matches..." 
However, you might just want to delete this test; I do not think it adds any value beyond that of the next test.